src/connector/src/sink/mqtt.rs

```diff
@@ -244,15 +248,45 @@ struct MqttSinkPayloadWriter {
     retain: bool,
 }
 
-impl FormattedSink for MqttSinkPayloadWriter {
-    type K = Vec<u8>;
-    type V = Vec<u8>;
+impl MqttSinkPayloadWriter {}
+
+impl MqttSinkPayloadWriter {
+    async fn write_chunk<F: SinkFormatter>(
+        &mut self,
+        chunk: StreamChunk,
+        formatter: &F,
+    ) -> Result<()>
+    where
+        F::K: SerTo<Vec<u8>>,
+        F::V: SerTo<Vec<u8>> + QueryField,
+    {
+        for r in formatter.format_chunk(&chunk) {
+            let (_, event_object) = r?;
+            let topic = {
+                if let Some(name) = &self.config.topic_field {
+                    match &event_object {
+                        Some(event_object) => {
+                            event_object.get_field(name).context("column not found")?
+                        }
+                        None => self.config.common.topic.clone(),
+                    }
+                } else {
+                    self.config.common.topic.clone()
```

Comment:
cloning a string each time is not a good practice, how about returning a ref? 

for example

```rust
let default_topic: &str = self.config.common.topic.as_ref();
for r in formatter.format_chunk(&chunk) {
    let (_, event_object) = r?;
    let topic: &str = {
            if let Some(name) = &self.config.topic_field {
                match &event_object {
                    Some(event_object) => {
                        event_object.get_field(name).context("column not found")?.as_ref()
                    }
                    None => default_topic,
                }
            } else {
                default_topic
            }
        };
    ...
}

```

Reply:
Oh thanks, this was a leftover from the original implementation, will fix it